modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/OpenAPIResourceImpl.java

```diff
@@ -53,10 +72,203 @@ public OpenAPIResourceImpl(
 	public Response getOpenAPI(@PathParam("type") String type)
 		throws Exception {
 
-		return _openAPIResource.getOpenAPI(
+		Response openAPI = _openAPIResource.getOpenAPI(
 			_openAPISchemaFilter, _resourceClasses, type, _uriInfo);
+
+		OpenAPI entity = (OpenAPI)openAPI.getEntity();
+
+		List<String> relationshipsEndpoints =
+			_removeRelationshipsOpenAPIEndpoints(
+				entity, _getRelationshipsEndpoints(entity.getPaths()),
+				endpoint -> endpoint.contains(_BATCH));
+
+		relationshipsEndpoints.forEach(
+			endpoint -> {
+				_objectRelationshipRelatedObjectDefinitionMap.forEach(
+					(objectRelationship, objectDefinition) ->
+						_createCustomRelationshipEndpointToOpenAPI(
+							entity, endpoint, objectRelationship,
+							objectDefinition));
+
+				Paths entityPaths = entity.getPaths();
+
+				entityPaths.remove(endpoint);
+			});
+
+		return openAPI;
+	}
+
+	private void _createCustomRelationshipEndpointToOpenAPI(
+		OpenAPI entity, String endpoint, ObjectRelationship objectRelationship,
+		ObjectDefinition relatedObjectDefinition) {
+
+		Paths entityPaths = entity.getPaths();
+
+		PathItem customRelationshipPathItem = _createCustomRelationshipPathItem(
+			entityPaths.get(endpoint), objectRelationship,
+			relatedObjectDefinition);
+
+		String currentObjectName = StringUtil.lowerCaseFirstLetter(
+			_currentObjectDefinition.getShortName());
+
+		String relatedObjectName = StringUtil.lowerCaseFirstLetter(
+			relatedObjectDefinition.getShortName());
+
+		String customRelationshipEndpoint = StringUtil.replace(
+			endpoint,
+			new String[] {
+				_CURRENT_OBJECT_ID, _RELATIONSHIP_NAME_IN_ENDPOINT,
+				_RELATED_OBJECT_ID
+			},
+			new String[] {
+				currentObjectName + "Id", objectRelationship.getName(),
+				relatedObjectName + "Id"
+			});
+
+		entityPaths.addPathItem(
+			customRelationshipEndpoint, customRelationshipPathItem);
+	}
+
+	private PathItem _createCustomRelationshipPathItem(
+		PathItem relationshipPathItem, ObjectRelationship objectRelationship,
+		ObjectDefinition relatedObjectDefinition) {
+
+		Operation relationshipPutOperation = relationshipPathItem.getPut();
+
+		List<Parameter> relationshipPutOperationParameters =
+			relationshipPutOperation.getParameters();
+
+		Stream<Parameter> relationshipPutOperationParametersStream =
+			relationshipPutOperationParameters.stream();
+
+		List<Parameter> customRelationshipPutOperationParameters =
+			relationshipPutOperationParametersStream.map(
```

Comment:
We can use the TransformUtil here as we have already the dependency with Vulcan and we have it there available to avoid the usage of Streams just to make a map to transform the source of the collection